File: 2023-2024/sourcecode/mod1/lab2/22SelectionBasicsJava/app/src/main/java/com/example/a22_selectionbasics_java/NumberListViewHolder.java
package com.example.a22_selectionbasics_java;

import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.selection.ItemDetailsLookup;
import androidx.recyclerview.widget.RecyclerView;

import com.example.a22_selectionbasics_java.databinding.RvItemBinding;

class NumberListViewHolder extends RecyclerView.ViewHolder {

    private final RvItemBinding binding;
    public NumberListViewHolder(RvItemBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public void bind(int value, boolean isActivated){
        binding.numberText.setText(String.valueOf(value));
        binding.getRoot().setActivated(isActivated);
    }

    public ItemDetailsLookup.ItemDetails<Long> getItemDetails() {
        return new ItemDetailsLookup.ItemDetails<Long>() {
            @Override
            public int getPosition() {
                return getAdapterPosition();
            }

            @NonNull
            @Override
            public Long getSelectionKey() {
                return getItemId();
            }
        };
    }
}

File: 2023-2024/sourcecode/mod1/lab2/21SelectorBasicsKotlin/app/src/main/java/com/example/a21_selectorbasics_kotlin/NumberListViewHolder.kt
package com.example.a21_selectorbasics_kotlin

import androidx.recyclerview.selection.ItemDetailsLookup
import androidx.recyclerview.widget.RecyclerView
import com.example.a21_selectorbasics_kotlin.databinding.ActivityMainBinding
import com.example.a21_selectorbasics_kotlin.databinding.RvItemBinding

class NumberListViewHolder(private val binding: RvItemBinding) :
    RecyclerView.ViewHolder(binding.root) {

    fun bind(value: Int, isActivated: Boolean = false) {
        binding.numberText.text = value.toString()
        binding.root.isActivated = isActivated
    }

    fun getItemDetails(): ItemDetailsLookup.ItemDetails<Long> =
        object : ItemDetailsLookup.ItemDetails<Long>() {
            override fun getPosition(): Int = adapterPosition
            override fun getSelectionKey(): Long = itemId
        }
}